### PR TITLE
Allow customization of plugins

### PR DESCRIPTION
**Pull request by tbetbetbe**

### CODE REVIEW

File: bin/grpc-api-packager [hunk 1]
```diff
@@ -69,29 +72,33 @@ var parseArgs = function parseArgs() {
...
+      nargs: '*',
+      dest: 'languages'
     }
   );
   cli.addArgument(
     [ '-r', '--services_root' ],
     {
       help: 'The directory containing the rpc service proto definitions.'
-          + '\nShould not be specified with -r/--protobuf, which clobbers this.'
+          + '\nShould not be specified with -r/--protobuf, which clobbers this.',
```
Comment (saicheems) [inline bin/grpc-api-packager]: Nit: some help statements followed by period, some don't...

File: bin/grpc-api-packager [hunk 2]
```diff
@@ -55,13 +56,15 @@ var parseArgs = function parseArgs() {
...
+        ' outside of an apis directory, but within services_root or repository',
+      dest: 'includePath'
     }
   );
   cli.addArgument(
-    [ '-o', '--out_path' ],
+    [ '-o', '--out_dir' ],
     {
-      help: 'The directory in which the generated packages will be saved.'
+      help: 'The directory in which the generated packages will be saved.',
```
Comment (saicheems) [inline bin/grpc-api-packager]: Same.

File: bin/grpc-api-packager [hunk 3]
```diff
@@ -116,7 +125,8 @@ var parseArgs = function parseArgs() {
       defaultValue: "",
```
Comment (saicheems) [inline bin/grpc-api-packager]: Single quotes?

File: bin/grpc-api-packager [hunk 4]
```diff
@@ -125,28 +135,49 @@ var parseArgs = function parseArgs() {
...
+      dest: 'overridePlugins'
     }
   );
   return cli.parseArgs();
 }
 
+var parsePlugins = function parsePlugins(pluginSpecs) {
+  var parsedSpecs = [];
+  _.forEach(pluginSpecs, function(p) {
+    var split = p.split(":");
```
Comment (saicheems) [inline bin/grpc-api-packager]: Single quotes?

File: lib/api_repo.js [hunk 5]
```diff
@@ -222,14 +223,19 @@ ApiRepo.prototype._checkDeps = function _checkDeps(opts, done) {
...
+      reqdBins.push(that.overridePlugins[l]);
+    } else {
+      reqdBins.push('grpc_' + l + '_plugin');
+    }
   });
 
-  var that = this;
   var isInPath = function isInPath(err, data) {
     if (!err) {
       var binPaths = data.split("\n");
```
Comment (saicheems) [inline lib/api_repo.js]: Single quotes?

File: bin/grpc-api-packager (hunk 1 above)
Comment (tbetbetbe) [inline bin/grpc-api-packager]: Fixed.  All help statements are now followed by a period.

File: bin/grpc-api-packager (hunk 2 above)
Comment (tbetbetbe) [inline bin/grpc-api-packager]: Fixed

File: bin/grpc-api-packager (hunk 3 above)
Comment (tbetbetbe) [inline bin/grpc-api-packager]: Fixed

File: lib/api_repo.js (hunk 5 above)
Comment (tbetbetbe) [inline lib/api_repo.js]: Fixed

**Pull request merged**